lib/compose.js

```diff
@@ -67,20 +92,20 @@ module.exports = function (ary, wrap) {
   }
 
   function parseMaybe (str) {
-    return  isString(str) ? parse(str) : str
+    return  isString(str) ? parse(str.split(';')[0]) : str
```

Comment:
I'm not sure about this behavior -- it seems like sometimes we're trying to parse the output of `stringify()`, but what happens when `stringify()` outputs multiple semicolon-delimited multiserver addresses?

Previously this code was crashing, now it's just parsing the first address passed. Is that okay?

Reply:
The case I've been dealing with is that we often do something like:

```javascript
const addr = combined.stringify('local')
combined.client(addr)
```

This has worked great in the past when `stringify()` only returns a single address, but when it returns multiple semicolon-delimited addresses then it fails. Should we force people to instead do:

```javascript
const addr = combined.stringify('local').split(';')[0]
```

?

Reply:
it should accept a multiaddress (with ;'s) it should connect to the first one that it understands. there is no gaurantee that the first one will work, maybe you are a browser and you can't do net addresses, etc.

Reply:
if there is something that isn't working with multiaddresses that should lets make sure that it does

Reply:
Thanks for taking the time to point this out and explain it to me. I didn't fully understand the expectations around `parse()` but I think I've got it now. I've just pushed 98b983e, which ensures that `parse()` can handle `;`-delimited multiserver addresses. Roughly, the behavior is:

- Split on `;`
  - Split up the parts with `SE.parse()`
    - Pass each part to its corresponding layer for parsing
  - Discard any parsed parts with falsy values (which indicate that the layer couldn't correctly parse)
  - Return the first correctly parsed address in our list, *or* return `null` to indicate that we don't have a way to parse that multiserver address.

Super appreciative about how many tests touch this code, they all fail when you get this implementation slightly wrong. :~)